src/components/pages/Publish/FormPublish.tsx

```diff
@@ -146,7 +150,13 @@ export default function FormPublish(): ReactElement {
               {...field}
               options={
                 field.type === 'boxSelection'
-                  ? accessTypeOptions
+                  ? accessTypeOptions.map((option) => {
+                      const value = values[field.name as keyof typeof values]
+                      return {
+                        ...option,
+                        checked: value && value === option.name
+                      }
+                    })
```

Comment:
question(blocking): Is there a way to extract this function as it is used at least twice in the same way? The issue is that someone might forget (or not know) that it needs to be updated in two files. Alternatively, we could add a comment in both files pointing to each other so everyone reading the code knows about it.

Reply:
Completely agree, I moved it to the InputElement component, this should simplify the maintenance of the component structure.